libraries/AP_HAL_ChibiOS/hwdef/common/flash.c

```diff
@@ -436,12 +438,18 @@ static bool stm32h7_flash_write32(uint32_t addr, const void *buf)
         SR = &FLASH->SR2;
     }
     stm32_flash_wait_idle();
+
     *CCR = ~0;
     *CR |= FLASH_CR_PG;
 
     const uint32_t *v = (const uint32_t *)buf;
-    uint8_t i;
-    for (i=0; i<8; i++) {
+    bool success = true;
```

Comment:
I think we want to do the checking of all the bytes before we write any, otherwise we're still doing a partial write. So I think we need a function check_all_ones() where we pass the address and length (in multiples of 4 bytes)
We can do the check before we unlock I think

Reply:
Fixed